database/model.js

```diff
@@ -1,11 +1,162 @@
+    )
+  `;
+
+  const query1 = await Promise.all(
+    departurePoints.map((departurePoint) =>
+      db
+        .query(SELECT_PLATFORM_EXIT_ID, [departurePoint])
+        .then((res) => res.rows)
+    )
+  );
+  // For hard coded we get [[], [{id for exit interchange}]]
+
+  const SELECT_PLATFORM_EXITS = /* SQL */ `
+    SELECT *
+    FROM platform_exits
+    WHERE platform_id = (
+      SELECT id 
+      FROM platforms 
+      WHERE individual_stop_id = $1
+    )
+  `;
+
+  let SELECT_INTERCHANGE;
+  let SELECT_EXIT;
+  const query2 = await Promise.all(
+    arrivalPoints
+      .map((arrivalPoint, index, arr) => {
+        if (index !== arr.length - 1) {
+          SELECT_INTERCHANGE = SELECT_PLATFORM_EXITS + `AND id = $2`;
+          return [arrivalPoint, query1[index + 1][0].platform_exit_id];
+        } else {
+          SELECT_EXIT = SELECT_PLATFORM_EXITS + `AND type = $2`;
+          return [arrivalPoint, 0];
+        }
+      })
+      .map((arrivalPoint, index, arr) => {
+        if (arr.length === index + 1) {
+          return db.query(SELECT_EXIT, arrivalPoint).then((res) => res.rows[0]);
+        } else {
+          return db
+            .query(SELECT_INTERCHANGE, [arrivalPoint[0], arrivalPoint[1]])
+            .then((res) => res.rows[0]);
+        }
+      })
+  );
+
+  return query2;
+}
+
+async function getStationNameByIndividualStopIds(stopIds) {
+  const SELECT_STATION_NAMES = /* SQL */ `
+    SELECT common_name_short 
+    FROM stations 
+    WHERE station_naptan = (
+      SELECT station_naptan
+      FROM platforms 
+      WHERE individual_stop_id = $1
+    )
+  `;
+
+  const stationNames = await Promise.all(
+    stopIds.map((stopId) => {
+      return db
+        .query(SELECT_STATION_NAMES, [stopId])
+        .then((res) => res.rows[0]);
+    })
+  );
+  return stationNames;
+}
+
+// async function getStation(stationNaptan) {
+//   const SELECT_STATION = `SELECT common_name_short FROM stations WHERE station_naptan=$1;`;
+//   const station = await db.query(SELECT_STATION, [stationNaptan]);
+//   return station.rows[0];
+// }
+
+// Takes an array of station naptans and returns an array of rows with one column (common_name_short)
+async function getStationCommonNamesFromNaptans(stationNaptans) {
+  const SELECT_STATIONS = /* SQL */ `
+    SELECT common_name_short
+    FROM stations
+    WHERE station_naptan = ANY ($1)
+    ORDER BY idx($1, station_naptan) -- Look at init.sql for what idx does
+  `;
+
+  const stationNames = await db.query(SELECT_STATIONS, [stationNaptans]);
+
+  return stationNames.rows;
+}
+
+async function getPlatformDataFromIndividualStopPoints(stopIds) {
+  const SELECT_PLATFORM_DATA = /* SQL */ `
+    SELECT platforms.tfl_public_direction_name AS line_direction, platforms.train_direction, lines.name AS line_name
+    FROM platforms, lines, platform_line
+    WHERE (lines.id = platform_line.line_id AND platforms.id = platform_line.platform_id)
+    AND platforms.individual_stop_id = ANY ($1)

```

Comment:
I think wherever we use ANY ($1) we will also want to use idx to order by the order of the array. I think we only caught it where the actual name was different as was easy to see, but we probably got the wrong data for each platform but didn't realise as we had really random data in it